packages/langium/src/serializer/json-serializer.ts

```diff
@@ -52,9 +60,12 @@ export class DefaultJsonSerializer implements JsonSerializer {
                 } else {
                     objects.add(item);
                 }
-                // If it is a reference, just return the name
+                // If it is a reference, transform it into a path
                 if (isReference(item)) {
-                    return { $refText: item.$refText } as Reference; // surprisingly this cast works at the time of writing, although $refNode is absent
+                    const refValue = item.ref;
+                    return {
+                        $ref: refValue && this.astNodeLocator.getAstNodePath(refValue)
```

Comment:
Proposal: prepend the path with `#` so we can later add support for references to other documents in the format `{uri}#{path}`